improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+
+# Diagnostics that differ from the PROB / PERC / DIAG pattern (not all are handled)
+ANCILLARIES = [
+    "surface_altitude",
+    "land_fraction",
+    "land_binary_mask",
+    "grid_with_halo",
+    "topographic_zone_weights",
+    "topography_mask",
+    "silhouette_roughness",
+    "standard_deviation_of_height_in_grid_cell",
+    "timezone_mask",
+    "smoothing_coefficient_x",
+    "smoothing_coefficient_y",
+]
+EMOS_COEFF_NAMES = [
+    f"emos_coefficient_{coeff}" for coeff in ["alpha", "beta", "gamma", "delta"]
+]
+INTERMEDIATES = [
+    "grid_neighbours",
+    "grid_eastward_wind",
+    "grid_northward_wind",
+    "precipitation_advection_x_velocity",
+    "precipitation_advection_y_velocity",
+    "reliability_calibration_table",
+] + EMOS_COEFF_NAMES
+
+SPECIAL_CASES = ["weather_code", "wind_from_direction"] + INTERMEDIATES + ANCILLARIES
+
+# Expected coordinates for different field types
+SPOT_COORDS = ["spot_index", "latitude", "longitude", "altitude", "wmo_id"]
+UNBLENDED_TIME_COORDS = ["time", "forecast_period", "forecast_reference_time"]
+BLENDED_TIME_COORDS = ["time", "blend_time"]
+LOCAL_TIME_COORDS = ["time", "time_in_local_timezone"]
+
+# Compliant, required and forbidden cell methods
+NONCOMP_CMS = [
+    CellMethod(method="mean", coords="forecast_reference_time"),
+    CellMethod(method="mean", coords="model_id"),
+    CellMethod(method="mean", coords="model_configuration"),
+    CellMethod(method="mean", coords="realization"),
+]
+NONCOMP_CM_METHODS = ["point", "weighted_mean"]
+COMP_CM_METHODS = ["min", "max", "minimum", "maximum", "sum"]
+PRECIP_ACCUM_CM = CellMethod(method="sum", coords="time")
+PRECIP_ACCUM_NAMES = [
+    "lwe_thickness_of_precipitation_amount",
+    "lwe_thickness_of_sleetfall_amount",
+    "lwe_thickness_of_snowfall_amount",
+    "thickness_of_rainfall_amount",
+]
+
+# Compliant, required and forbidden attributes
+NONCOMP_ATTRS = [
+    "mosg__grid_type",
+    "mosg__grid_domain",
+    "mosg__grid_version",
+    "mosg__forecast_run_duration",
+    "grid_id",
+    "source_realizations",
+    "um_version",
+]
+DIAG_ATTRS = {
+    "weather_code": ["weather_code", "weather_code_meaning"],
+    "wind_gust": ["wind_gust_diagnostic"],
+}
+COMP_ATTRS = MANDATORY_ATTRIBUTES + [
+    "Conventions",
+    "least_significant_digit",
+    "mosg__model_configuration",
+]
+
+# Expected substrings to be found in certain title attributes
+BLEND_TITLE_SUBSTR = "IMPROVER Post-Processed Multi-Model Blend"
+PP_TITLE_SUBSTR = "Post-Processed"
+SPOT_TITLE_SUBSTR = "Spot Values"
+
+
+class MOMetadataInterpreter:
+    """Class to interpret an iris cube according to the Met Office specific
+    IMPROVER standard.  This is intended as a debugging tool to aid developers
+    in adding and modifying metadata within the code base."""
+
+    PROB = "probabilities"
+    PERC = "percentiles"
+    DIAG = "realizations"
+    ANCIL = "ancillary"
+
+    def __init__(self):
+        """Initialise class parameters"""
```

Comment:
Might be useful to note that this stores information about the cube which is then displayed by the display function. I got confused and was going to ask why some of these things were stored on self w hen they weren't used by this class, but they are used by the display function.

Reply:
Good idea - have expanded this docstring.

Reply:
Thanks